RHDEVDOCS-6101: DOC: Add a flag on controllers to configure resyncPeriod in openshift-pipelines (#79107)
openshift/openshift-docs: modules/op-setting-resync-period.adoc

// This module is included in the following assemblies:
// * install_config/customizing-configurations-in-the-tektonconfig-cr.adoc

:_mod-docs-content-type: CONCEPT
[id="op-setting-resync-period_{context}"]

> = Setting resync period for {pipelines-shortname}

Reviewer: ```suggestion
= Setting the resync period for the pipelines controller
```

> You can set a resync period for {pipelines-shortname}, which allows the controller to reconcile all pipeline runs and task runs, regardless of events. Setting a resync period is typically done if the default 10-hour timeframe is too short and costly for the number of objects being reconciled.

Reviewer: ```suggestion
You can configure the resync period for the pipelines controller. Once every resync period, the controller reconciles all pipeline runs and task runs, regardless of events. The default resync period is 10 hours. If you have a large number of pipeline runs and task runs, a full reconciliation every 10 hours might consume too many resources. In this case, you can configure a longer resync period.
```

> .Example of setting resync period for {pipelines-shortname} to 24 hours

Reviewer: ```suggestion
.Example of setting the resync period for the pipelines controller to 24 hours
```

// This module is included in the following assemblies:
// * install_config/customizing-configurations-in-the-tektonconfig-cr.adoc

:_mod-docs-content-type: PROCEDURE
[id="op-setting-resync-period_{context}"]
= Setting the resync period for the pipelines controller

You can configure the resync period for the pipelines controller. Once every resync period, the controller reconciles all pipeline runs and task runs, regardless of events. 

The default resync period is 10 hours. If you have a large number of pipeline runs and task runs, a full reconciliation every 10 hours might consume too many resources. In this case, you can configure a longer resync period.

.Prerequisites
* You are logged in to your {OCP} cluster with `cluster-admin` privileges.

.Procedure

. In the `TektonConfig` CR, configure the resync period for the pipelines controller, as shown in the following example.
+

> .Example of setting the resync period for the pipelines controller to 24 hours

Reviewer 2: Suggestion, so this is up to you - would any of these shorter descriptions still work?
```suggestion
.Example `TektonConfig` CR with 24 hour resync period
```
```suggestion
.Example pipelines controller with 24 hour resync period
```
Or simply
```suggestion
.Example
```

Up to you but I think a lot of the context is already explained in the procedure and the file itself, imo

[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
spec:
  pipeline:
    options:
      deployments:
        tekton-pipelines-controller:
          spec:
            template:
              spec:
                containers:
                - name: tekton-pipelines-controller
                  args:

>                     - "-resync-period=24h"

Reviewer 2: Suggestion - since you don't mention which specific value to modify in the actual step, would it be helpful for users if you added a callout label here and explained the variable underneath the example?
```suggestion
                    - "-resync-period=24h" <1>
```
